Cinecon/VisitorsMenu.cs

```diff
@@ -274,5 +275,126 @@ private static void ShowItemTypes(MenuCategory category, string menuItem)
+                if (invalidEmail)
+                    ConsoleHelper.ColorWriteLine("   Voer a.u.b. een valide e-mail adres in.\n", ConsoleColor.Red);
+                email = ConsoleHelper.ReadLineWithText("   Wat is jouw e-mail adres? -> ", writeLine: false);
+                invalidEmail = true;
+            }
+            while (!emailRegex.Match(email).Success);
+
+            Console.Clear();
+
+            var paymentMethods = new Dictionary<string, Action>();
+
+            foreach (var method in JsonHelper.ReservationData.PaymentMethods)
+                paymentMethods[method] = null;
+
+            var paymentMethodChoice = new ChoiceMenu(paymentMethods, true, "   Kies een betaalmethode\n").MakeChoice();
+
+            if (paymentMethodChoice.Key == "Terug")
+            {
+                // TODO: Back to seat selection.
+            }
+            else
+                FinishPaymentProcess(movie, paymentMethodChoice.Key, name, email);
+        }
+
+        private static void FinishPaymentProcess(Movie movie, string paymentMethod, string name, string email)
+        {
+            ConsoleHelper.LogoType = LogoType.Films;
+            ConsoleHelper.Breadcrumb = $"Films / {movie.Title} / Koop tickets / Betaling";
+
+            var confirmationChoice = ChoiceMenu.CreateConfirmationChoiceMenu(
+                $"   Is de ingevulde data correct?\n\n   Film: {movie.Title}\n   Naam: {name}\n   E-mail: {email}\n   Betaalmethode: {paymentMethod}\n{MenuCartText}\n").MakeChoice();
+
+            if (confirmationChoice.Key != "Ja")
+            {
+                StartPaymentProcess(movie);
+                return;
+            }
+
+            var reservation = new Reservation
+            {
+                Name = name,
+                Email = email,
+                Code = GenerateRandomCode(),
+                PaymentMethod = paymentMethod,
+                IsActivated = false,
+                Movie = movie,
+                Seats = new List<Seat>()
+            };
+
+            JsonHelper.ReservationData.Reservations.Add(reservation);
+
+            JsonHelper.UpdateJsonFiles();
+
+            Task.Run(() => SendReservationEmail(reservation));
+
+            new ChoiceMenu(new Dictionary<string, Action>
+            {
+                { "Afronden", ShowVisitorMenu }
+            }, text: $"   Jouw betaling met {paymentMethod} is succesvol afgerond. Jouw reserveringscode is {reservation.Code}.\n   Je zult straks ook een e-mail krijgen met jouw reserveringscode.\n   " +
+                $"Bedankt en tot de voorstelling!\n").MakeChoice().Value();
+        }
+
+        private static string GenerateRandomCode()
+        {
+            var randomCode = "";
+            var r = new Random();
+            var chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+
+            do
+            {
+                randomCode = new string(Enumerable.Range(1, 5).Select(_ => chars[r.Next(chars.Length)]).ToArray());
```

Comment:
Waarom laat je de code niet altijd beginnen met een letter of juist met een getal zodat het beter sorteerbaar is.

Reply:
Bij het sorteren staan getallen voor letters, dus uiteindelijk maakt het niet uit. Het wordt prima gesorteerd.